_posts/blog/2016-09-13-code-reviews.md

```diff
@@ -0,0 +1,70 @@
+# Consult. Dont preach.
+
+The basic mindset in which an engineer should approach a code review for a change introduced by another engineer is the mindset of a consultant. Whatever you say or write: do it with caution. Sentences that start with "I would approach the problem this way..." or "I believe this way would be more efficient" are a good approach if you want to encourage the reviewee to accept your recommendations. Way better than "this solution is inappropriate" or "do it like this". Command-style comments are guaranteed to result in damage. Conflicts, drop of creativity and motivation, bad team spirit. Dont do it.
+By the way: adding "please" to such a sentence doesnt help. You already clearly communicated a difference in status. You already switches your consultant-hat for something that is simply inappropriate in modern software development teams. The result will be a bad taste in the mouth at best.
+By the way: the (probably correct) advice will now not be accepted full-heated. And team morale is damaged.
+
+If the reviewer would have invested just a little more time and watched the way he communicated: the reviewee might have been very thankful and might have accepted all advise with a smile.
+
+Your effort is well invested if you take a break from time to time and make sure your mindset is tuned, your communication style is respectful and reflects that you are a buddy who just wants to help.
+
+
+# Do not enforce consensus.
+
+Engineers often tend to forget that there is more than one valid opinion in a lot of cases. Not everything is determined to false and true.
+So if the reviewer has communicated his opinion appropriate and was not able to convince the reviewee, the code review is the wrong forum to let discussion escalate and enforce consensus. If the code is robust and fulfills the requirements, let it pass the review and try to move the topic out of the situation. You may have tech meetings or other forums in which you can discuss the topic (ideally in general and not by specific example) with a broader audience.
+Letting the topic lie for a week can also help. Often times you realize that consensus isnt important at all. That not only there are more valid opinions than yours. But also that your team maybe can afford to let both opinions stand.
+
+# Do not point at rules but give explanations
+
+Rules and processes kill creativity. You want to give as much freedom to everyone as possible without hurting someone else (thats part of the german constitution by the way ;)).
+Developers develop processes and methods. They align on conventions. And other developers violate those. Often times even consceniously. How to handle those violations?
+Often times now is the time when the reviewer changes from consultant to finger-pointing judge (and hangman) who points at the wiki in which he documented the rule three years ago (without ever discussing it properly). Usually the tone gets rough (catch some breath, adjust your mindset...).
```

Comment:
a lot of stuff in brackets in this paragraph. how about using commas or connecting words.